### PR TITLE
Release 0.1 docs

### PR DESCRIPTION
**Pull request by zaleslaw**
KDocs for release

### CODE REVIEW

**Comment by zaleslaw** [comment]
@MKhalusova please have a look on the KDocs in the proposed PR

**Review by MKhalusova — APPROVED** [review]
Overall, the documentation looks good and detailed enough. I left a few comments, and I did not check for minor grammar issues. I can do another pass at it later, or we can involve someone else for such task.

File: api/src/main/kotlin/api/inference/InferenceModel.kt [hunk 1]
```diff
@@ -20,32 +20,50 @@ import java.io.File
...
         set(value) {
             (underlyingLogger as Logger).level = value
         }
 
-    open fun predict(image: FloatArray): Int {
-        reshape(image).use { tensor ->
+    /**
+     * Predicts the class of [inputData].
+     *
+     * @param [inputData] The single example (or observation) with unknown label.
```
[the remark this answers is not in the document]
Comment (zaleslaw) [reply api/src/main/kotlin/api/inference/InferenceModel.kt:62]: I'd prefer observation term

File: api/src/main/kotlin/api/keras/activations/Activations.kt [hunk 2]
```diff
@@ -4,23 +4,217 @@ import api.keras.util.getDType
 import org.tensorflow.Operand
 import org.tensorflow.op.Ops
 
+/**
+ * Neural network hyper-parameter, activation function of a node defines the output of that node given an input or set of inputs.
```
[the remark this answers is not in the document]
Comment (zaleslaw) [reply api/src/main/kotlin/api/keras/activations/Activations.kt:8]: I was in doubts

File: api/src/main/kotlin/api/keras/callbacks/Callback.kt [hunk 3]
```diff
@@ -3,128 +3,125 @@ package api.keras.callbacks
...
 
     /**
-     * Called at the end of a training batch in `fit` methods. This method is
-     * empty. Extend this class to handle this event.
+     * Called at the end of a batch during training phase.
      *
-     * @param batch index of batch within the current epoch.
-     * @param event Metric results for this batch.
+     * @param [batch] index of batch within the current epoch.
+     * @param [batchSize] Amount of observation in the current batch.
```
[the remark this answers is not in the document]
Comment (zaleslaw) [reply api/src/main/kotlin/api/keras/callbacks/Callback.kt:48]: Agree

File: api/src/main/kotlin/api/keras/TrainableModel.kt [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+import org.tensorflow.Operand
+
+/**
+ * Base abstract class for all trainable models.
+ */
+abstract class TrainableModel : InferenceModel() {
+    /** Controls level of verbosity. */
+    protected var isDebugMode = false
+
+    /** Optimizer. Approach how aggressively to update the weights. */
```
[the remark this answers is not in the document]
Comment (zaleslaw) [reply api/src/main/kotlin/api/keras/TrainableModel.kt:22]: Need to think about

**Comment by zaleslaw** [comment]
So, @MKhalusova I'll continue my work and 'll try to be more consistent

**Pull request merged**